core/shared/src/main/scala/zio/logging/internal/LogAppender.scala

```diff
@@ -46,16 +46,11 @@ private[logging] trait LogAppender { self =>
   /**
    * Appends a key/value string pair to the log.
    */
-  final def appendKeyValue(key: String, value: String): Unit = appendKeyValue(key, _.appendText(value))
-
-  /**
-   * Appends a key/value pair, with the value it created with the log appender.
-   */
-  final def appendKeyValue(key: String, appendValue: LogAppender => Unit): Unit = {
+  def appendKeyValue(key: String, value: String): Unit = {
```

Comment:
Let's not modify the `appendKeyValue`. Instead, generate an anonymous `LogAppender` that will simply produce string for any possible combination of operations.

Reply:
@jdegoes: previously removed function def appendKeyValue(key: String, appendValue: LogAppender => Unit) of LogAppender- returned back